src/checkers/inference/solver/backend/lingeling/LingelingSolver.java

```diff
@@ -33,7 +33,10 @@
 public class LingelingSolver extends MaxSatSolver {
 
     // Lingeling binary executable file should be located at JSR308/lingeling/lingeling.
-    private final String lingeling = System.getenv().get("JSR308") + "/lingeling/lingeling";
+    private String currentDirPath = System.getProperty("user.dir");
+    private String JSR308 = currentDirPath.substring(0, currentDirPath.lastIndexOf("checker-framework-inference/"));
+
+    private final String lingeling = JSR308 + "/lingeling/lingeling";
```

Comment:
One idea that might make this even less dependent on JSR308 (or the common root directory structure) is to put lingeling under CFI/lib, just like how z3 is currently set up, see https://github.com/opprop/checker-framework-inference/blob/master/scripts/buildZ3

the lingeling setup script is here https://github.com/opprop/checker-framework-inference/blob/master/scripts/buildLingeling

if you also think this is a good idea, feel free to make the edits (to the setup script, eclipse classpath file, and to build.gradle to point to the new location of lingeling binaries)

if you feel its too much trouble and work then what you have in the PR is okay. let me know what you'd prefer.

Reply:
Hi Jeff, I think it's a good idea. Since I am a beginner, I am not sure that I can finish it successfully, maybe it will take me some time, but I will try to do this.

Reply:
Ok lets make that a TODO. I'll file an enhancement issue for you to track.